pytensor/tensor/subtensor.py

```diff
@@ -178,16 +215,51 @@ def as_index_literal(
             as_index_literal(idx.step),
         )
 
+    if not isinstance(idx, Variable):
+        raise NotScalarConstantError()
+
+    if isinstance(idx.type, NoneTypeT):
+        return None
+
+    if isinstance(idx, ScalarConstant):
+        return cast(int, idx.data)
+
+    if isinstance(idx, TensorConstant):
+        return cast(int, idx.data.item())
+
+    if isinstance(idx, SliceConstant):
+        return cast(slice, idx.data)
+
+    if isinstance(idx.type, SliceType):
+        assert idx.owner is not None
+        return slice(*map(as_index_literal, idx.owner.inputs))
+
+    # Other kinds of variables are not supported
     raise NotScalarConstantError()
 
 
 def get_idx_list(inputs, idx_list):
     return indices_from_subtensor(inputs[1:], idx_list)
 
 
+@overload
+def get_canonical_form_slice(
+    theslice: slice | SliceConstant,
+    length: int | np.integer | ScalarVariable | TensorVariable,
+) -> tuple[slice, int | ScalarConstant]: ...
+
+
+@overload
+def get_canonical_form_slice(
+    theslice: ScalarVariable | TensorVariable,
+    length: int | np.integer | ScalarVariable | TensorVariable,
+) -> tuple[ScalarVariable, int]: ...
+
+
 def get_canonical_form_slice(
```

Comment:
Feel free to give it a more verbose name of you want, rather than extend the behavior to handle slice types from advanced indexing operations.

Or raise explicitly for SliceType/NonType that this is not mean to handle those types. I really don't see a reason to extend unused functionality in a refactor / type hint PR

```suggestion
 def get_canonical_form_basic_subtensor_slice(
```